tests/trainer/test_trainer.py

```diff
@@ -110,17 +111,32 @@ class TestTrainerEquivalence():
     reference_folder: pathlib.Path
 
     def assert_models_equal(self, model_1, model_2):
+        assert model_1 is not model_2, "Same model should not be compared."
         for param1, param2 in zip(model_1.parameters(), model_2.parameters()):
-            torch.testing.assert_allclose(param1, param2)
+            torch.testing.assert_allclose(param1, param2, **self.threshold)
+
+    @pytest.fixture(autouse=True)
+    def threshold(self, device, precision, world_size):
+        self.threshold = {'atol': 0, 'rtol': 0}
```

Comment:
if this works, this is awesome!

Reply:
Yeah, I've identified that it doesn't work in some situations (e.g. mixed precision, and also curiously gradient accumulation on CPU is off by rtol=1e-8), but doing some testing to set strict thresholds where possible.